packages/plugins/Export/src/store/actions.ts

```diff
@@ -1,7 +1,92 @@
 import { jsPDF } from 'jspdf'
 import { PolarActionTree } from '@polar/lib-custom-types'
+import { Map } from 'ol'
 import { ExportFormat, ExportGetters, ExportState } from '../types'
 
+// PDF options
+const dims = {
+  a0: [1189, 841],
+  a1: [841, 594],
+  a2: [594, 420],
+  a3: [420, 297],
+  a4: [297, 210],
+  a5: [210, 148],
+}
+
+// Screenshot canvas
+const CANVAS_ID = 'export-canvas'
+
+const convertToPdf = (map: Map, src: string, size: number[]) => {
+  // NOTE: when supporting more formats, scale map accordingly
+  const format = 'a4'
+  const dim = dims[format]
+  // Import of jspdf is in mounted.
+  const jsPdf = new jsPDF('landscape', undefined, format) // eslint-disable-line
+  jsPdf.addImage(src, 'JPEG', 0, 0, dim[0], dim[1])
+  // Reset original map size
+  map.setSize(size)
+  map.getView().setResolution(map.getView().getResolution())
+
+  return {
+    pdfSrc: jsPdf.output('datauristring'),
+    jsPdf,
+  }
+}
+
+const downloadAsImage = (src: string, type: ExportFormat) => {
+  const link = document.createElement('a')
+  link.download = 'map.' + (type === ExportFormat.PNG ? 'png' : 'jpeg')
+  link.href = src
+  document.body.appendChild(link)
+  link.click()
+  document.body.removeChild(link)
+}
+
+// For every ol-layer, get matrix and apply on canvas
+const getDataUrl = (size: number[], type: ExportFormat) => {
+  // The canvas to render the screenshot in
+  const mapCanvas: HTMLCanvasElement = document.createElement('canvas')
+  mapCanvas.id = CANVAS_ID
+  // mapCanvas.setAttribute('crossOrigin', 'anonymous')
+  mapCanvas.width = size[0]
+  mapCanvas.height = size[1]
+
+  // Write on this
+  const mapContext = mapCanvas.getContext('2d')
+  if (!mapContext) {
+    console.error(
+      '@polar/plugin-export: map does not of a 2d context, export failed.'
```

Comment:
```suggestion
      '@polar/plugin-export: map does not have a 2d context, export failed.'
```
Let's fix this while we are at it